packages/headless/src/controllers/quickview/headless-quickview.ts

```diff
@@ -33,7 +33,7 @@ export interface Quickview extends Controller {
   /**
    * Retrieves the preview content for the configured result.
    */
-  fetchResultContent(): void;
+  fetchResultContent(preventAnalytics: boolean): void;
```

Comment:
We cannot add a required param because it will be a breaking change to devs upgrading.

Suggestion:
- Rename this controller to`headless-core-quickview` and remove the analytics
- Create a wrapper for search that sends the search analytics.
- Create a wrapper for case-assist that sends the case-assist analytics.

Reply:
Thanks for the suggestion, I'll update my code accordingly👌

Reply:
I applied your suggestion and created the headless-core-quickview and the wrapper that will be used when building a quickview for search and that will take care of sending the search analytics, but I wasn't able to create a same wrapper for case-assist cause to send the case-assist analytics we need the case assist engine and in the quickview controller we build the controller using a search engine and not a case-assist one.
The reason why the case assist analytic actions need to be dispatched using a case assist engine is because in this actions we need to gather some data from the state of the case assist engine. for example when we want to send a suggestion_click event, which is sent when clicking on a document suggestion, we dispatch the action `logDocumentSuggestionClick` with one parameter which is the id of this suggestion,  in the body of this action we need to retrieve all the data of this suggestion specified with this id parameter in order to send the data of this suggestion in the analytics, and we retrieve this data from the state of the case assist engine. so when dispatching this same action with a search engine the document suggestion will never be found since the state of the search engine doesn't contain any document suggestions.
So now when using the quickview in a case-assist use case I'm building the quickview with the headless-core-quickview, to avoid sending any analytics, and then I'm handling sending the right analytics using the case assist engine from the quantic side.
What do u think about that @samisayegh ?

Reply:
What if we adjust the type of the core quickview controller to accept a `CoreEngine`? The dev facing wrapper controller would continue to use `CaseAssistEngine`. Logging this event under the hood would be nice 🤞